tests/Unit/DataStructures/Test_Variables.cpp

```diff
@@ -510,69 +510,128 @@ void test_variables_prefix_semantics() noexcept {
+  using Prefix1VariablesType = Variables<
+      tmpl::list<VariablesTestTags_detail::PrefixTag1<
+                     VariablesTestTags_detail::tensor_tag<VectorType>>,
+                 VariablesTestTags_detail::PrefixTag1<
+                     VariablesTestTags_detail::scalar_tag<VectorType>>,
+                 VariablesTestTags_detail::PrefixTag1<
+                     VariablesTestTags_detail::scalar2_tag<VectorType>>>>;
+  using Prefix2VariablesType = Variables<
+      tmpl::list<VariablesTestTags_detail::PrefixTag2<
+                     VariablesTestTags_detail::tensor_tag<VectorType>>,
+                 VariablesTestTags_detail::PrefixTag2<
+                     VariablesTestTags_detail::scalar_tag<VectorType>>,
+                 VariablesTestTags_detail::PrefixTag2<
+                     VariablesTestTags_detail::scalar2_tag<VectorType>>>>;
+  using Prefix3VariablesType = Variables<
+      tmpl::list<VariablesTestTags_detail::PrefixTag3<
+                     VariablesTestTags_detail::tensor_tag<VectorType>>,
+                 VariablesTestTags_detail::PrefixTag3<
+                     VariablesTestTags_detail::scalar_tag<VectorType>>,
+                 VariablesTestTags_detail::PrefixTag3<
+                     VariablesTestTags_detail::scalar2_tag<VectorType>>>>;
+  using value_type = typename VectorType::value_type;
+
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<tt::get_fundamental_type_t<value_type>> dist{1,
+                                                                         100};
+  UniformCustomDistribution<size_t> sdist{5, 20};
+
+  const size_t number_of_grid_points = sdist(gen);
+  const auto value_in_variables = make_with_random_values<value_type>(
+      make_not_null(&gen), make_not_null(&dist));
+  const std::array<value_type, 3> random_vals =
+      make_with_random_values<std::array<value_type, 3>>(make_not_null(&gen),
+                                                         make_not_null(&dist));
+
+  // Test arithmetic operators +, -, *, / for prefix Variables
+  Prefix0VariablesType prefix_vars0(number_of_grid_points, value_in_variables);
+  Prefix1VariablesType prefix_vars1(number_of_grid_points, value_in_variables);
+  Prefix2VariablesType prefix_vars2(number_of_grid_points, value_in_variables);
+  Prefix3VariablesType prefix_vars3(number_of_grid_points, value_in_variables);
+
+  check_vectors(Prefix3VariablesType{number_of_grid_points,
+                                     value_in_variables * random_vals.at(0)},
+                random_vals.at(0) * prefix_vars0);
+  check_vectors(Prefix3VariablesType{number_of_grid_points,
+                                     value_in_variables * random_vals.at(0)},
+                prefix_vars0 * random_vals.at(0));
+  check_vectors(Prefix3VariablesType{number_of_grid_points,
+                                     value_in_variables / random_vals.at(0)},
+                prefix_vars0 / random_vals.at(0));
+  check_vectors(
```

Comment:
explain why we might want to ever add things of different prefixes. @nilsdeppe @wthrowe @kidder Why do we want to do this?